cdap-app-fabric/src/main/java/co/cask/cdap/app/store/ServiceStore.java

```diff
@@ -39,4 +39,25 @@
    * @param instances Instance Count.
    */
   void setServiceInstance(String serviceName, int instances);
+
+  /**
+   * Update the service instances restart request.
+   *
+   * @param serviceName Service name to be restarted.
+   * @param startTime Start time in Ms from Epoch.
+   * @param endTime End time in Ms from Epoch.
+   * @param isSuccess Whether or not the operation successful.
+   * @param instanceId The instance Id to be restarted. If null then it restart all instances.
+   */
+  void setRestartInstancesRequest(String serviceName, long startTime, long endTime, boolean isSuccess,
```

Comment:
Seems weird to restart all if `instanceId` is null. Why not have two methods, `requestInstanceRestart` and `requestAllInstancesRestart`?